src/debug/createdump/memoryregion.h

```diff
@@ -36,23 +47,57 @@ struct MemoryRegion
         assert((end & ~PAGE_MASK) == 0);
     }
 
-    const uint32_t Permissions() const { return m_permissions; }
+    // copy with new file name constructor
+    MemoryRegion(const MemoryRegion& region, char* fileName) : 
+        m_flags(region.m_flags),
+        m_startAddress(region.m_startAddress),
+        m_endAddress(region.m_endAddress),
+        m_offset(region.m_offset),
+        m_fileName(fileName)
+    {
+    }
+
+    // copy with new flags constructor. The file name is not copied.
+    MemoryRegion(const MemoryRegion& region, uint32_t flags) : 
+        m_flags(flags),
+        m_startAddress(region.m_startAddress),
+        m_endAddress(region.m_endAddress),
+        m_offset(region.m_offset),
+        m_fileName(nullptr)
+    {
+    }
+
+    // copy constructor
+    MemoryRegion(const MemoryRegion& region) : 
+        m_flags(region.m_flags),
+        m_startAddress(region.m_startAddress),
+        m_endAddress(region.m_endAddress),
+        m_offset(region.m_offset),
+        m_fileName(region.m_fileName)
+    {
+    }
+
+    const uint32_t Permissions() const { return m_flags & MEMORY_REGION_FLAG_PERMISSIONS_MASK; }
+    const uint32_t Flags() const { return m_flags; }
+    const bool IsBackedByMemory() const { return (m_flags & MEMORY_REGION_FLAG_MEMORY_BACKED) != 0; }
     const uint64_t StartAddress() const { return m_startAddress; }
     const uint64_t EndAddress() const { return m_endAddress; }
     const uint64_t Size() const { return m_endAddress - m_startAddress; }
     const uint64_t Offset() const { return m_offset; }
-    const char* FileName() const { return m_fileName; }
+    char* FileName() const { return m_fileName; }
```

Comment:
I wonder why was this change needed. There doesn't seem to be any newly added call to this method in the current PR and I assume the original version with the `const char*` return was ok before.